### PR TITLE
Look up position and add as leg if valid

### PR DESCRIPTION
**Pull request by elizabeth-jimenez**
Dual Merge: 
- [API PR ](https://github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/698)
- [Mock PR](https://github.com/MetaPhase-Consulting/mock-fsbid/pull/291)

### CODE REVIEW

File: src/sass/_agendaItemMaintenance.scss [hunk 1]
```diff
@@ -387,10 +387,8 @@
 
   .leg-dropdown {
     background-image: none;
+    border: 2px solid transparent;
     border-bottom: 2px solid $color-black;
-    border-left: none;
-    border-right: none;
-    border-top: none;
```
Comment (elizabeth-jimenez) [inline src/sass/_agendaItemMaintenance.scss:393]: @scott062 change requested in other PR :D

File: src/sass/_agendaItemMaintenance.scss [hunk 2]
```diff
@@ -272,14 +272,34 @@
...
         height: 28px;
         padding: 0 30px 0 5px;
         max-width: 150px;
         font-size: 13px;
       }
+
+      .loading-animation {
+        //background: linear-gradient(135deg, white 0% 30%, $primary-alt-light 30% 35%, white 35% 100%) right;
+        background: linear-gradient(135deg, white 0% 30%, $primary-alt-light 30% 70%, white 70% 100%) right;
+        //background: linear-gradient(135deg, white 0%, $primary-alt-light 50%, white 100%) right;
```
[the remark this answers is not in the document]
Comment (SidelineCory24) [reply src/sass/_agendaItemMaintenance.scss:285]: I like the current option😎

[the remark this answers is not in the document]
Comment (elizabeth-jimenez) [reply src/sass/_agendaItemMaintenance.scss:285]: That was my favorite one too 😀

**Comment by elizabeth-jimenez** [comment]
I stashed my updates to tests, to prioritize reviewing PRs

File: src/Components/Agenda/AgendaItemMaintenancePane/AgendaItemMaintenancePane.jsx [hunk 3]
```diff
@@ -46,20 +45,43 @@ const AgendaItemMaintenancePane = (props) => {
...
+
+  useDidMountEffect(() => {
+    const pos = get(pos_results, 'results[0]');
+
+    if (pos === undefined) {
+      setPosNumError(true);
+    } else {
+      setPositionNumber('');
+    }
+  }, [pos_results]);
```
Comment (scott062) [inline src/Components/Agenda/AgendaItemMaintenancePane/AgendaItemMaintenancePane.jsx:72]: Sorry why do we need this vs just using the error loading state for position search?

Comment (elizabeth-jimenez) [reply src/Components/Agenda/AgendaItemMaintenancePane/AgendaItemMaintenancePane.jsx:72]: Error isn't triggered on empty - this handles empty

File: src/Components/Agenda/AgendaItemTimeline/AgendaItemTimeline.jsx [hunk 4]
```diff
@@ -27,98 +29,54 @@ const AgendaItemTimeline = ({ unitedLoading, setParentState }) => {
...
       setLocalLoading(false);
       setParentState(false);
     }, '9000');
   }, []);
 
+  useEffect(() => {
+    if (!pos_results_loading) {
+      const pos = get(pos_results, 'results[0]');
+      if (pos) {
+        const legs = selectedLegs;
```
Comment (scott062) [inline src/Components/Agenda/AgendaItemTimeline/AgendaItemTimeline.jsx:53]: Actual copy?

Comment (elizabeth-jimenez) [reply src/Components/Agenda/AgendaItemTimeline/AgendaItemTimeline.jsx]: ahh good point! Spread it

File: src/Components/Agenda/AgendaItemMaintenancePane/AgendaItemMaintenancePane.jsx [hunk 5]
```diff
@@ -46,20 +45,43 @@ const AgendaItemMaintenancePane = (props) => {
...
+
   const [selectedPositionNumber, setPositionNumber] = useState();
+  const [posNumError, setPosNumError] = useState(false);
+
   const [selectedPanelCat, setPanelCat] = useState(get(panelCategories, '[0].mic_code'));
   const [selectedPanelMLDate, setPanelMLDate] = useState();
   const [selectedPanelIDDate, setPanelIDDate] = useState();
 
+  useEffect(() => {
+    setParentState(includes([asgSepBidLoading,
```
Comment (scott062) [inline src/Components/Agenda/AgendaItemMaintenancePane/AgendaItemMaintenancePane.jsx:57]: rename setParentLoadingState ? Or something to describe what parent state its passing up

File: src/Components/Agenda/AgendaItemMaintenancePane/AgendaItemMaintenancePane.jsx (hunk 3 above)
Comment (scott062) [reply src/Components/Agenda/AgendaItemMaintenancePane/AgendaItemMaintenancePane.jsx:72]: My thought here is that useSelector for positionSearchHasErrored and a length check would cover it right? Thinking through three scenarios:
1) Finds position (Success) (Length > 0)
2) Finds no positions (Success) - But nothing to add - How do we handle this? Error or No results instead? (Length < 1)
3) Errors - Endpoint Failure - True failure that the search just didn't work (Error state?)

Just thinking we want a way to tell the user the difference between no results and an error

File: src/Components/Agenda/AgendaItemMaintenancePane/AgendaItemMaintenancePane.jsx (hunk 5 above)
Comment (elizabeth-jimenez) [reply src/Components/Agenda/AgendaItemMaintenancePane/AgendaItemMaintenancePane.jsx]: done 😁

**Review by scott062 — APPROVED** [review]
💪 🐐

**Pull request merged**